src/api/verification.py

```diff
@@ -0,0 +1,170 @@
+
+    def configure(
+        self,
+        output_path: str = None,
+        lib_items_path: str = None,
+        plot_option: str = None,
+        fig_size: tuple = None,
+        num_threads: int = 1,
+        preprocessed_data: pd.DataFrame = None,
+    ) -> None:
+        """Configure verification environment.
+
+        Args:
+            output_path (str): Verification results output path.
+            lib_items_path (str): Verification library path (include name of the file with extension).
+            plot_option (str): Type of plots to include. It should either be all-compact, all-expand, day-compact, or day-expand. It can also be None, which will plot all types.
+            fig_size (tuple): Tuple of integers (length, height) describing the size of the figure to plot.
+            num_threads (int): Number of threads to run verifications in parallel. Defaults to 1.
+            preprocessed_data (pd.DataFrame, optional): Pre-processed data stored in the data frame. Optional.
+        """
+        if self.cases is None or len(self.cases) == 0:
+            logging.error(
+                "The verification case suite is empty, there is nothing to configure."
+            )
+            return None
+        if output_path is None:
+            logging.error("An output_path argument should be specified.")
+            return None
+        elif not os.path.isdir(output_path):
+            logging.error("The specificed output directory does not exist.")
+            return None
+
+        if lib_items_path is None:
+            logging.error(
+                "A path to the library of verification cases should be provided."
+            )
+            return None
+        elif not isinstance(lib_items_path, str):
+            logging.error("The path to the library of verification cases is not valid.")
+            return None
+        elif not os.path.isfile(lib_items_path):
+            logging.error("The path to the library of verification cases is not valid.")
+            return None
+        elif "json" != lib_items_path.split(".")[-1].lower():
+            logging.error("The library should be a JSON file.")
+            return None
+
+        if not plot_option in [
+            None,
+            "all-compact",
+            "all-expand",
+            "day-compact",
+            "day-expand",
+        ]:
+            logging.error(
+                f"The plot_option argument should either be all-compact, all-expand, day-compact, or day-expand, not {plot_option}."
+            )
+
+        if isinstance(fig_size, tuple):
+            if not (
+                (isinstance(fig_size[0], int) or isinstance(fig_size[0], float))
+                and (isinstance(fig_size[1], int) or isinstance(fig_size[1], float))
+            ):
+                logging.error(
+                    "The fig_size argument should be a tuple of integers or floats."
+                )
+                return None
+        elif not fig_size is None:
```

Comment:
does fig_size take None as a valid value? seems like run_libcase will handle it with default size. 
If so, I probably would want the default to be explicitly shown in the api. so I am going to change this to not handle None, and give fig_size here a default value of (6.4, 4.8)

Reply:
Yes, `fig_size` can take `None`.